enumerable.rb

```diff
@@ -0,0 +1,169 @@
+      temp = []
+      value.my_each do |a|
+        temp.push(a) if yield(a)
+      end
+      temp
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(arg = nil)
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return false unless yield(a, b) }
+      else
+        value.my_each { |a| return false unless yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return false unless a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return false unless a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return false unless a }
+    end
+    true
+  end
+
+  def my_any?
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return true if yield(a, b) }
+      else
+        value.my_each { |a| return true if yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return true if a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return true if a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return true if a }
+    end
+    false
+  end
+
+  def my_none?
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return false if yield(a, b) }
+      else
+        value.my_each { |a| return false if yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return false if a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return false if a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return false if a }
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    value = self
+    count = 0
+    if block_given?
+      value.my_each do |a|
+        count += 1 if yield(a)
+      end
+    end
+
+    unless arg.nil?
+      value.my_each do |a|
+        count += 1 if arg == a
+      end
+    end
+
+    count = value.length if !block_given? && arg.nil?
+    count
+  end
+
+  def my_map(&block)
+    arr = []
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| arr.push(block.call(a, b)) }
+      elsif value.is_a? Array
+        value.my_each { |a| arr.push(block.call(a)) }
+      end
+    else
+      to_enum(:my_map)
+    end
+    arr
+  end
+
+  def my_inject(init = nil, arg = nil, &block)
+    value = self
+    sum
+    if block_given?
+      sum = init.nil? ? value[0] : init
+      value.my_each_with_index { |a, i| sum = block.call(sum, a) unless i.zero? }
+    elsif !init.nil? && arg.nil?
+      if init.is_a? Symbol
+        sum = value[0]
+        value.my_each_with_index { |a, i| sum = sum.send(init, a) unless i.zero? }
+      else
+        sum = init
+        value.my_each { |a| sum = block.call(sum, a) }
+      end
+    elsif !init.nil? && !arg.nil?
+      sum = init
+      value.my_each { |a| sum = sum.send(arg, a) }
+    end
+    sum
+  end
+
+  def multiply_els
+    my_inject { |sum, result| sum + result }
+  end
```

Comment:
Let's use the multiply `*` operator instead of addition `+` operator, the method is about multiplication, not addition.

Let's put `multiply_els` outside the `Enumerable` module. This will allow us to use `multiply_els` as described in the [Odin assignment page](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2) as shown below:

```ruby
multiply_els([2,4,5]) #=> 40
```